core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -668,7 +669,7 @@ public void defaultCheckout() throws IOException, InterruptedException {
                     }
                 } catch (AbortException e) {
                     listener.error(e.getMessage());
-                } catch (InterruptedIOException e) {
+                } catch (ClosedByInterruptException | InterruptedIOException e) {
```

Comment:
The change to `SCM.java` elsewhere in this PR (from traditional Java I/O to NIO.2) means that cancelling a job while SCM checkout is taking place can now throw `ClosedByInterruptException` as well as `InterruptedException` and `InterruptedIOException`. Higher-level code marks the build as `ABORTED` for `InterruptedException` and `FAILURE` for other I/O exceptions, so we convert the `ClosedByInterruptException` to an `InterruptedException` here (just as the existing code does for `InterruptedIOException`) so that the build will be shown as `ABORTED` (as with the pre-NIO.2 logic).